docs/research/user-experience/xmr-btc-user-interface-definition.md

```diff
@@ -0,0 +1,145 @@
+# XMR<>BTC Atomic Swap - User Interface Prototypes
+
+This document specifies assumptions concerning possible user interfaces on top of the [Monero-Bitcoin swap as described by Lucas on the COMIT-blog](https://comit.network/blog/2020/10/06/monero-bitcoin/).
+
+We first specify assumptions and limitations imposed by the current setup and software needed for the swap, then outline possible solutions. 
+This document will be used as a basis for defining and evaluating low-fidelity UI prototypes in the prototyping tool Figma.
+
+## Assumptions
+
+This section sums up assumptions around the current setup, protocol and existing software. 
+
+### Technical Limitations
+
+Swap-solutions that run completely in the browser (webpage) can be problematic because blockchain nodes, notably [bitcoind](https://github.com/bitcoin/bitcoin/issues/11833), do not specifying [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS) in their response.
+This means the browser will reject a response from a locally running Bitcoin node.
+
+Theoretically this could be overcome by creating a node infrastructure behind a proper REST API, or with a locally running proxy. However, it is questionable if these are favourable solution, as they adds additional software on top of the blockchain nodes.  
+We will focus on solutions that favour the use of locally running blockchain nodes.
+
+### Current Setup
+
+To be able to provide a reasonably trustless setup, where the user is in control of his private keys the user is currently advised to run his own Bitcoin and Monero node.
+
+Given the current setup this results in the following software necessary to achieve a swap: 
+
+* Bitcoin Blockchain Node
+* Monero Blockchain Node
+* Bitcoin Wallet
+* Monero Wallet
+* **Swap Execution Daemon**: Monitor the blockchain and move the swap forward to the next action.
+* **Interface to connect the trading partners**: The two parties have to find each other and start the `Swap Execution Daemon` with the correct parameters. The interface can be realised in multiple different ways, from single-maker, P2P trading platform to P2P marketplace. 
+
+For Bitcoin the current setup relies on a synced bitcoind node and bitcoind's wallet. 
+
+For Monero the current setup relies on a synced monerod node and monero's wallet RPC. For funding Monero an existing wallet is unlocked and used for the swap. The current setup reliese on generating a *new* Monero wallet for `redeeming` or `refunding`.
+This is not optimal for the user experience, as it means that a user cannot redeem or refund to an existing Monero wallet in the current setup. We decided **not** to model this restriction of the current setup in the prototypes, but to model the prototypes using an existin Monero wallet. 
```

Comment:
Note: there is a relatively simple solution to that problem: we can temporarily create a new wallet for redeem/refund key and spend the output directly to an address from the user's wallet. This means the protocol has one more on-chain transaction but you can "ignore" this problematic.